src/topology/omega_complete_partial_order.lean

```diff
@@ -0,0 +1,212 @@
+preserves the joins of ω-chains  -/
+@[reducible]
+def Scott (α : Type u) := α
+
+instance scott_topological_space (α : Type u) [preorder α] : topological_space (Scott α) :=
+{ is_open := scott_topological_space.is_open α,
+  is_open_univ := scott_topological_space.is_open_univ α,
+  is_open_inter := scott_topological_space.is_open_inter α,
+  is_open_sUnion := scott_topological_space.is_open_sUnion α }
+
+@[simp]
+lemma le_iff_imp {p q : Prop} : p ≤ q ↔ (p → q) := by refl
+
+section
+variables {α : Type*} [preorder α] (y : Scott α)
+
+/-- `not_below` is an open set in `Scott α` used
+to prove the monotonicity of continuous functions -/
+def not_below := { x | ¬ x ≤ y }
+
+lemma f_below_is_open : is_open (not_below y) :=
+begin
+  have h : monotone (not_below y),
+  { intros x y' h,
+    simp only [not_below, set_of, le_iff_imp],
+    intros h₀ h₁, apply h₀ (le_trans h h₁) },
+  existsi h,
+  rintros c x ⟨h₀,h₁⟩, split,
+  { intros i,
+    simp only [not_below, set_of, le_iff_imp, preorder_hom.coe_fun_mk, chain.map_to_fun, function.comp_app],
+    intros h₂ h₃, apply h₂,
+    transitivity, apply h₀, exact h₃ },
+  { simp only [not_below, set_of, le_iff_imp, preorder_hom.coe_fun_mk, chain.map_to_fun, function.comp_app],
+    intros p hp hyx,
+    by_contradiction h₂, apply hyx, clear hyx,
+    apply h₁, intro i,
+    by_contradiction h₃,
+    apply h₂ (hp _ h₃), }
+end
+
+end
+
+open scott_topological_space (hiding is_open)
+open omega_complete_partial_order
+
+@[simp]
+lemma set_of_apply {α} (p : α → Prop) (x : α) : set_of p x ↔ p x := iff.rfl
+
+lemma is_ωSup_ωSup {α} [omega_complete_partial_order α] (c : chain α) :
+  is_ωSup c (ωSup c) :=
+begin
+  split,
+  { apply le_ωSup, },
+  { apply ωSup_le, },
+end
+
+lemma Scott_continuous_of_continuous {α β}
+  [omega_complete_partial_order α]
+  [omega_complete_partial_order β]
+  (f : Scott α → Scott β) (hf : continuous f) :
+  omega_complete_partial_order.continuous' f :=
+begin
+  have h : monotone f,
+  { intros x y h,
+    cases (hf {x | ¬ x ≤ f y} (f_below_is_open _)) with hf hf', clear hf',
+    specialize hf h, simp only [set.preimage, set_of, (∈), set.mem, le_iff_imp] at hf,
+    by_contradiction, apply hf a (le_refl (f y)) },
+  existsi h, intro c,
+  apply eq_of_forall_ge_iff, intro z,
+  simp only [ωSup_le_iff, preorder_hom.coe_fun_mk, chain.map_to_fun, function.comp_app],
+  specialize (hf _ (f_below_is_open z)),
+  cases hf, specialize hf_h c (ωSup c) _,
+  simp only [not_below, set.preimage, set.mem_set_of_eq, set_of_apply] at hf_h,
+  { cases hf_h with h₀ h₁, split, intros,
+    { transitivity; [skip, apply a],
+      apply h, apply le_ωSup, },
+    { simp only [le_iff_imp, preorder_hom.coe_fun_mk, chain.map_to_fun, function.comp_app, set_of_apply] at h₀ h₁,
+      intros h₂,
+      by_contradiction h₃,
+      refine h₁ false _ h₃,
+      intros i h₄, apply h₄ (h₂ i), } },
+  { split, apply le_ωSup,
+    apply ωSup_le, }
+end
+
+lemma continuous_of_Scott_continuous {α β}
```

Comment:
If you defined an open set literally as the preimage of `True` by an order-continuous function `X -> Prop`, then this theorem should just be an application of the fact that order-continuous functions are closed under composition.